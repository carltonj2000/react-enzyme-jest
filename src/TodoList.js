import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import { addTodo, removeTodo } from "./store";

export class TodoList extends Component {
  state = { input: "" };
  handleClick = i => () => {
    this.props.removeTodo(i);
  };
  handleChange = e => {
    this.setState({ input: e.currentTarget.value });
  };
  handleSubmit = () => {
    this.props.addTodo({ text: this.state.input });
    this.setState({ input: "" });
  };
  render() {
    return (
      <Fragment>
        <ul>
          {this.props.todos.map(({ text }, i) => (
            <li key={i} onClick={this.handleClick(i)}>
              {text}
            </li>
          ))}
        </ul>
        <input onChange={this.handleChange} value={this.state.input} />
        <button onClick={this.handleSubmit}>Add Todo</button>
      </Fragment>
    );
  }
}

const mapStateToProps = ({ currentList: { todos } }) => ({ todos });

const bindActionsToDispatch = dispatch => ({
  addTodo: todo => dispatch(addTodo(todo)),
  removeTodo: id => dispatch(removeTodo(id))
});

const TodoListContainer = connect(mapStateToProps, bindActionsToDispatch)(
  TodoList
);

export default TodoListContainer;
