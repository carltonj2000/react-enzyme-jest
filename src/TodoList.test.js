import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

import { TodoList } from "./TodoList";

configure({ adapter: new Adapter() });

describe("<TodoList />", () => {
  it("calls addTodo Redux action", () => {
    const props = { addTodo: jest.fn(), todos: [] };
    const wrapper = shallow(<TodoList {...props} />);
    wrapper
      .find("input")
      .simulate("change", { currentTarget: { value: "Buy milk" } });
    wrapper.find("button").simulate("click");
    expect(props.addTodo).toHaveBeenCalledWith({ text: "Buy milk" });
  });
  it("calls removeTodo Redux action", () => {
    const props = { removeTodo: jest.fn(), todos: ["hi", "there"] };
    const wrapper = shallow(<TodoList {...props} />);
    wrapper
      .find("li")
      .at(1)
      .simulate("click");
    expect(props.removeTodo).toHaveBeenCalledWith(1);
  });
  it("matches snapshot", () => {
    const props = { todos: ["hi", "there", "folks"] };
    const wrapper = shallow(<TodoList {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
