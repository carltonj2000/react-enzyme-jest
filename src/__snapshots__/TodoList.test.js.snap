// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> matches snapshot 1`] = `
<React.Fragment>
  <ul>
    <li
      key="0"
      onClick={[Function]}
    />
    <li
      key="1"
      onClick={[Function]}
    />
    <li
      key="2"
      onClick={[Function]}
    />
  </ul>
  <input
    onChange={[Function]}
    value=""
  />
  <button
    onClick={[Function]}
  >
    Add Todo
  </button>
</React.Fragment>
`;
